### PR TITLE
Coloring stream

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This is an initial pull request in case someone wants to do some review.

I'll be updating this over the next day or three with further updates.

### CODE REVIEW

File: sources/lib/coloring-stream/ansi-coloring-stream.dylan [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    (attributes :: <text-attributes>)
+ => (ansi-codes :: <byte-string>)
+  let fg = attributes.text-foreground-color;
+  let bg = attributes.text-background-color;
+  let intensity = attributes.text-intensity;
+  with-output-to-string(s)
+    write(s, "\<1b>[");
+    let need-separator = #f;
+    if (fg)
+      write(s, integer-to-string(30 + fg.color-ansi-code));
```
Comment (cgay) [inline sources/lib/coloring-stream/ansi-coloring-stream.dylan]: What are 30 and 40?  Worth using constants?

File: documentation/library-reference/source/coloring-stream/index.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+     Instances of ``<text-attributes>`` are used for representing the
+     desired text appearance. They can be passed to :gf:`format` when
+     writing to a :class:`<coloring-stream>`.
+
+     *background* and *foreground*, if given, should be one of the color
+     constants like :const:`$color-red`, :const:`$color-green`, etc.
+
+     *intensity*, if given, should be one of :const:`$bright-text-intensity`,
+     :const:`$dim-color-intensity` or :const:`$normal-text-intensity`.
```
Comment (cgay) [inline documentation/library-reference/source/coloring-stream/index.rst]: should be ``$dim-text-intensity``
(two occurrences)

File: documentation/library-reference/source/coloring-stream/index.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+   :description:
+
+     .. note:: Not all terminals support dimmed text.
+
+.. constant:: $normal-text-intensity
+
+Text Colors
+-----------
+
+.. constant:: $color-black
```
Comment (cgay) [inline documentation/library-reference/source/coloring-stream/index.rst]: What about leaving off the "color-" prefix? People can always rename on import in the unlikely case they're also defining ``$red`` etc.

Comment (waywardmonkeys) [inline documentation/library-reference/source/coloring-stream/index.rst]: I've been thinking about this as well. One argument that is either for or against, I'm not sure which, is that DUIM and Deuce have constants named ``$red`` and so on.

File: documentation/library-reference/source/coloring-stream/index.rst (hunk 2 above)
Comment (waywardmonkeys) [inline documentation/library-reference/source/coloring-stream/index.rst]: I fixed this after your comment, but then I decided to rename the constants to ``$dim-intensity`` and so on.

**Comment by waywardmonkeys** [comment]
I've updated this with some (more) bikeshedding.

I've also included a prototype commit that colorizes compiler warnings. (Clearly a work in progress.)

**Pull request merged**